### PR TITLE
feat: Add Ultracube, closes #1465

### LINKED ISSUE #1465: [mod request] Ultracube
**Need it right now?**
Check out the [Wiki](https://github.com/factoriolab/factoriolab/wiki/Creating-a-new-mod-set) to see how to build the mod data and update the code yourself, and create a pull request or run the project locally!

**Have you checked whether this mod or set of mods has already been requested?**
yes

**What mod set are you requesting?**Ultracube

**What mods are included in this mod set?**
Ultracube

### PR DESCRIPTION
**Pull request by Dimava**
note: Plz fix dupe modules, I didn't add them
image

### CODE REVIEW

**Comment by dcbroad3** [comment]
How on earth did you generate that so fast

File: src/data/index.ts [hunk 1]
```diff
@@ -84,6 +85,7 @@ export const data: AppData = {
     'pysalf',
     'sea',
     'sxp',
+    'ulc',
```
Comment (dcbroad3) [inline src/data/index.ts:88]: This array is append-only, you have to add new ids at the end

**Comment by Dimava** [comment]
@dcbroad3 open cmd next to factorio.exe, paste 3 commands, mkdir in data so the next run factorio-buld doesn't crash, run factorio-build, run generate-hash, all with Bun cuz I hate npm

**Comment by dcbroad3** [comment]
> @dcbroad3 open cmd next to factorio.exe, paste 3 commands, mkdir in data so the next run factorio-buld doesn't crash, run factorio-build, run generate-hash, all with Bun cuz I hate npm

That's interesting. I wonder if the mod structure has changed, because when I tried that before I got all vanilla items and only one mod item.

**Comment by Dimava** [comment]
@dcbroad3 ah, you mean the mod
Yep, I did make a mod that enabled all cube-* techs

**Comment by Dimava** [comment]
```lua
-- data-final-fixes.lua
local function __TS__StringStartsWith(self, searchString, position)
    if position == nil or position < 0 then
        position = 0
    end
    return string.sub(self, position + 1, #searchString + position) == searchString
end

for _, tech in pairs(data.raw.technology) do
    if __TS__StringStartsWith(tech.name, "cube-") then
        tech.enabled = true
    end
end
```

**Comment by Dimava** [comment]
Modules are still duped, dunno how to fix that as json doesn's have them

**Comment by dcbroad3** [comment]
I'll look into it later

**Comment by Dimava** [comment]
> I'll look into it later
@dcbroad3 you may merge this then
Just check that commit jsons have `efficiency-module` but don't have `"efficiency-module` (meaning all modules there are `"cube-efficiency-module` and thus it's not a json bug)

**Comment by dcbroad3** [comment]
Should we keep the "tuned / overcharged" versions of the modules and remove the original "module, module 2, module 3" versions, like this?
image

**Comment by Dimava** [comment]
> Should we keep the "tuned / overcharged" versions of the modules and remove the original "module, module 2, module 3" versions, like this?
Yes, regular modules are inaccessible and have different stats

**Comment by dcbroad3** [comment]
OK - I can't actually modify your PR myself, since it's in your repository. Here's the changes that are still necessary.

* Remove the "speed-module", "productivity-module", and "effectivity-module" _items_ (1-3) of each. The icon entries should be left alone, only the "items" should be removed. 
* Add a "defaults" property to the end of the json. This would work, based on the 1.1 defaults:

```json
"defaults":{"beacon":"beacon","minBelt":"transport-belt","maxBelt":"express-transport-belt","fuelRank":["coal"],"cargoWagon":"cargo-wagon","fluidWagon":"fluid-wagon","excludedRecipes":[],"minMachineRank":["assembling-machine-1","electric-furnace","electric-mining-drill"],"maxMachineRank":["assembling-machine-3","electric-furnace","electric-mining-drill"],"moduleRank":["cube-productivity-module-v3","cube-speed-module-v3"],"beaconModule":"speed-module-3"}
```
* Remove the "speed-module" _limitation_ and any references to it (`"limitation":"speed-module"). I'm not sure why this was added to the data set, since it's an empty limitation.

**Comment by Dimava** [comment]
> OK - I can't actually modify your PR myself, since it's in your repository.

No you can 
image

**Comment by Dimava** [comment]
IDK what I did but it seems fixed now

**Comment by dcbroad3** [comment]
> > OK - I can't actually modify your PR myself, since it's in your repository.
> 
> No you can image

Huh. I'm not sure how to do it then. I checked out the pr with the github cli and made a commit, but when I went to commit it acted like I was creating a new branch. If it's fixed it's fixed though

**Comment by dcbroad3** [comment]
Ah, it still needs the defaults entry to be added. I'll see if I can figure out how to actually commit to this pr tonight

**Pull request merged**